addon/services/toast.ts

```diff
@@ -0,0 +1,129 @@
+import { later } from '@ember/runloop';
+import Service from '@ember/service';
+import { isEmpty } from '@ember/utils';
+
+const TIMEOUT = 5000;
+
+enum ToastType {
+  INFO = 'info',
+  WARNING = 'warning',
+  ERROR = 'error',
+  SUCCESS = 'success'
+}
+
+export type ToastOptions = {
+  closeButton?: true;
+  tapToDismiss?: boolean;
+  onclick?: Function | null;
+};
+
+type ToastFunction = (message: string, title: string, opts: ToastOptions) => void;
+
+const DEFAULT_OPTIONS: ToastOptions = Object.freeze({
+  closeButton: true,
+  onclick: null,
+  tapToDismiss: true
+});
+
+function delegate(ctx: Toast, type: ToastType): ToastFunction {
+  return (message: string, title: string, opts: ToastOptions = DEFAULT_OPTIONS): void => {
+    if (isEmpty(message) && isEmpty(title)) return;
+
+    ctx.buildToast(type, message, title, { ...DEFAULT_OPTIONS, ...opts });
+  };
+}
+
+export default class Toast extends Service {
+  private _toasts: Element[] = [];
+
+  success: ToastFunction = delegate(this, ToastType.SUCCESS);
+  error: ToastFunction = delegate(this, ToastType.ERROR);
+  warning: ToastFunction = delegate(this, ToastType.WARNING);
+  info: ToastFunction = delegate(this, ToastType.INFO);
+
+  buildToast(type: ToastType, message: string, title: string, opts: ToastOptions): void {
```

Comment:
One last thing (apart from tests ! 😂 )
This method is a bit too long and can be split a bit more.

For instance, you create titleContainer & messageContainer, then apply the same logic underneath both of them. Why not have a createElementWithClassAndText method ? (rename it ofc ^^)

The closeButton logic in the if() can also be moved to a separate method.